### DISCUSSION TITLE
SET TIMEZONE Postgres Views

### DISCUSSION THREAD

**Discussion by ethanstan** [hasura/graphql-engine#7447] [Question]
What is the best way to utilize a user's local timezone when querying a postgres view through graphql?

**Comment by leoalves** [comment]
Convert the local timezone to UTC before sending to the server.  And convert from UTC to the local timezone when displaying to the user

**Reply by ethanstan** [reply]
This is what we do for normal queries, however on a postgres view we have a group by day. Since some times will be marked with a different day depending on the user's local timezone our charts are showing incorrect data for that timezone.

**Reply by leoalves** [reply]
In your view you could convert the date field to the local timezone.  Using the saved user local timezone from the database.  Or changing the view to a function and passing the local timezone in the function. 
But if you start doing that in the server you will need to do it everywhere you have filters/aggregation in the date fields. It might get complex. 

I would avoid if possible, because Hasura doesn't support timezone conversion by default. And if you need it you will either have to create a view or function

```SQL
select '2021-08-24 06:38:46'::timestamp at time zone 'america/toronto'
```
